packages/cashier/src/Components/Error/__tests__/no-balance.spec.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { screen, render, fireEvent, cleanup } from '@testing-library/react';
+import { createBrowserHistory } from 'history';
+import { Router } from 'react-router';
+import { routes } from '@deriv/shared';
+import NoBalance from '../no-balance';
+
+jest.mock('Stores/connect', () => ({
+    __esModule: true,
+    default: 'mockedDefaultExport',
+    connect: () => Component => Component,
+}));
+
+describe('<NoBalance />', () => {
+    const history = createBrowserHistory();
+
+    it('component should render', () => {
+        render(
+            <Router history={history}>
+                <NoBalance is_deposit_locked={false} />
+            </Router>
+        );
+
+        expect(screen.getByRole('heading')).toBeInTheDocument();
+        expect(screen.getByText('Deposit now')).toBeInTheDocument();
+        expect(screen.getByText('Please make a deposit to use this feature.')).toBeInTheDocument();
+    });
+
+    it('must not able to make a deposit when deposit is locked', async () => {
+        render(
+            <Router history={history}>
+                <NoBalance is_deposit_locked />
+            </Router>
+        );
+
+        expect(screen.getByRole('heading')).toBeInTheDocument();
+        expect(screen.queryByText('Deposit now')).not.toBeInTheDocument();
+        expect(screen.queryByText('Please make a deposit to use this feature.')).not.toBeInTheDocument();
+    });
+
+    it('component should redirect to deposit page when button is clicked', () => {
+        const setTabIndex = jest.fn();
+        const cashier_deposit = 'cashier_deposit';
+
+        render(
+            <Router history={history}>
+                <NoBalance is_deposit_locked={false} setTabIndex={setTabIndex} />
+            </Router>
+        );
+
+        const btn = screen.getByRole('button');
+        fireEvent.click(btn);
+
+        expect(history.location.pathname).toBe(routes[cashier_deposit]);
```

Comment:
I think there is no need to declare `cashier_deposit` variable. You may call `routes.cashier_deposit` here.

Reply:
right, thanks for notice!